Tests/Service/Request/Visitor/RemoveNullFieldsRequestVisitorTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+declare(strict_types=1);
+
+namespace Auto1\ServiceAPIClientBundle\Tests\Service;
+
+use Auto1\ServiceAPIClientBundle\Service\Request\Visitor\EndpointConfiguration;
+use Auto1\ServiceAPIClientBundle\Service\Request\Visitor\RemoveNullFieldsRequestVisitor;
+use Http\Message\StreamFactory;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Class RemoveNullFieldsRequestVisitorTest.
+ */
+class RemoveNullFieldsRequestVisitorTest extends TestCase
+{
+    /**
+     * @var RequestInterface|ObjectProphecy
+     */
+    private $requestProphecy;
+
+    /**
+     * @var StreamFactory|ObjectProphecy
+     */
+    private $streamFactoryProfecy;
+
+    /**
+     * @var UriInterface|ObjectProphecy
+     */
+    private $uriInterfaceProfecy;
+
+    /**
+     * @var StreamInterface|ObjectProphecy
+     */
+    private $streamInterfaceProfecy;
+
+    protected function setUp()
+    {
+        $this->requestProphecy = $this->prophesize(RequestInterface::class);
+        $this->streamFactoryProfecy = $this->prophesize(StreamFactory::class);
+        $this->uriInterfaceProfecy = $this->prophesize(UriInterface::class);
+        $this->streamInterfaceProfecy = $this->prophesize(StreamInterface::class);
+    }
+
+    /**
+     * @dataProvider provider
+     */
+    public function testRemoveNullFields($input, $expected)
+    {
+        $this->requestProphecy->__call('withBody', [Argument::type(StreamInterface::class)])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->requestProphecy);
+
+        $this->requestProphecy->__call('getMethod', [])
+            ->willReturn("PUT");
+
+        $this->requestProphecy->__call('getUri', [])
+            ->willReturn($this->uriInterfaceProfecy);
+
+        $this->uriInterfaceProfecy->__call('getPath', [])
+            ->willReturn("/v2/vehicles/43");
+
+        $this->streamInterfaceProfecy->__call('getContents', [])
+            ->willReturn($input);
+        $this->streamInterfaceProfecy->reveal();
+
+        $this->requestProphecy->__call('getBody', [])
+            ->willReturn($this->streamInterfaceProfecy);
+
+        $this->streamFactoryProfecy->__call('createStream', [Argument::is($expected)])
+            ->willReturn($this->streamInterfaceProfecy);
+
+        $request = $this->requestProphecy->reveal();
+        $factory = $this->streamFactoryProfecy->reveal();
+        $removeNullFieldsRequestVisitor = new RemoveNullFieldsRequestVisitor(
+            $factory,
+            [
+                new EndpointConfiguration("PUT", "/v2/vehicles/{id}")
+            ]
+        );
+
+        $removeNullFieldsRequestVisitor->visit($request);
+    }
+
+    public function provider()
+    {
+        return [
```

Comment:
What will be if all elements will be removed?

Reply:
In case of all the elements are deleted the parent element will remain empty.

Reply:
And this empty element will be encoded into `{}` or `[]`? 
It's a quite important question for none-PHP services

Reply:
Will be encoded as `[]`

Reply:
Same question about empty assoc array. 

Could you cover these corner cases with tests also?